### PR TITLE
Updated UI for Effects list.

### PR DESCRIPTION
**Pull request by designedbyscience**
Also, added Blendmodes to effects

Created a new component for general BlendMode listing, and then renamed the old BlendMode component to be LayerBlendMode

Added a little popover dialog instead of using the Select or Datalist component for the action of adding a new effect

### CODE REVIEW

File: src/js/actions/layereffects.js [hunk 1]
```diff
@@ -333,6 +333,22 @@ define(function (require, exports) {
...
     setShadowSpread.reads = [locks.PS_DOC, locks.JS_DOC];
     setShadowSpread.writes = [locks.PS_DOC, locks.JS_DOC];
+    
+    /**
+     * Set the Drop Shadow Blend Mode value for all selected layers
+     *
+     * @param {Document} document
+     * @param {Immutable.Iterable.<Layer>} layers list of layers to update
+     * @param {number} shadowIndex index of the Drop Shadow within the layer(s)
+     * @param {string} blendMode  blendMode name
```
Comment (mcilroyc) [inline src/js/actions/layereffects.js]: missing `type` JSDoc

File: src/js/actions/layereffects.js [hunk 2]
```diff
@@ -333,6 +333,22 @@ define(function (require, exports) {
...
+     * @return {Promise}
+     */
+    var setShadowBlendMode = function (document, layers, shadowIndex, blendMode, type) {
+        return _upsertShadowProperties.call(
+           this, document, layers, shadowIndex, { blendMode: blendMode }, null, type);
+    };
+    setShadowBlendMode.reads = [locks.PS_DOC, locks.JS_DOC];
+    setShadowBlendMode.writes = [locks.PS_DOC, locks.JS_DOC];
```
Comment (mcilroyc) [inline src/js/actions/layereffects.js]: one school of thought says that you oughtn't include read locks for which you are specifying the associated write lock.  If this goes against the general pattern elsewhere in this file, then disregard

Comment (designedbyscience) [inline src/js/actions/layereffects.js]: Disregarding!

File: src/js/jsx/sections/style/Shadow.jsx [hunk 3]
```diff
@@ -239,7 +252,8 @@ define(function (require, exports) {
                 xPositions: collection.pluck(shadows, "x"),
                 yPositions: collection.pluck(shadows, "y"),
                 blurs: collection.pluck(shadows, "blur"),
-                spreads: collection.pluck(shadows, "spread")
+                spreads: collection.pluck(shadows, "spread"),
+                blendMode: collection.pluck(shadows, "blendMode")
```
Comment (mcilroyc) [inline src/js/jsx/sections/style/Shadow.jsx]: nit: for consistency this should probably be `blendModes`

File: src/js/jsx/sections/style/Shadow.jsx [hunk 4]
```diff
@@ -285,133 +299,123 @@ define(function (require, exports) {
...
-                                size="column-2"
-                            />
+                                onColorChange={this._opaqueColorChanged}
+                                onAlphaChange={this._alphaChanged}
+                                swatchOverlay={shadowOverlay}>
+                            </ColorInput>
                         </div>
-
-                        <Gutter />
+                        <div className="control-group column-23">
```
Comment (mcilroyc) [inline src/js/jsx/sections/style/Shadow.jsx]: too much indentation for this line?

File: src/js/jsx/sections/style/Shadow.jsx [hunk 5]
```diff
@@ -471,32 +475,23 @@ define(function (require, exports) {
...
+            if (shadowsContent.isEmpty()) {
+                return null;
+            }
 
             return (
-                <div className="shadow-list__container">
-                    <header className="shadow-list__header sub-header">
-                        <h3>
+                <div className="shadow-list__container ">
+                    <header className="shadow-list__header section-header section-header__no-padding">
```
Comment (mcilroyc) [inline src/js/jsx/sections/style/Shadow.jsx]: nit: this smells like it should be simpler.  you can safely ignore this comment.  There's probably a reason.

File: src/js/jsx/sections/style/Type.jsx [hunk 6]
```diff
@@ -543,7 +543,7 @@ define(function (require, exports, module) {
                                 swatchOverlay={typeOverlay} />
                         </div>
                         <div className="control-group__vertical control-group__no-label">
-                            <BlendMode
+                            <LayerBlendMode
                                     document={this.props.document}
```
Comment (mcilroyc) [inline src/js/jsx/sections/style/Type.jsx]: indentation :space_invader:

File: src/js/jsx/sections/style/VectorFill.jsx [hunk 7]
```diff
@@ -130,7 +130,7 @@ define(function (require, exports, module) {
                             fill={this.state.fill} />
                     </div>
                     <div className="control-group__vertical control-group__no-label">
-                        <BlendMode
+                        <LayerBlendMode
                                 id={this.props.id}
```
Comment (mcilroyc) [inline src/js/jsx/sections/style/VectorFill.jsx]: :space_invader:

**Comment by mcilroyc** [comment]
a few nits.  otherwise looks good.  sweet popover.  back to @designedbyscience

**Comment by designedbyscience** [comment]
Okay, thank you (and you were right about cleaning up the header, some extra classes there). Rebased and back to you @mcilroyc

**Pull request merged**